lib/runtime/wasmer/exports.go

```diff
@@ -227,6 +228,58 @@ func (in *Instance) QueryCallFeeDetails(ext []byte) (*types.FeeDetails, error) {
 // TODO: use this in block verification process (#1873)
 func (in *Instance) CheckInherents() {}
 
+// GrandpaGenerateKeyOwnershipProof returns grandpa key ownership proof from runtime.
+func (in *Instance) GrandpaGenerateKeyOwnershipProof(authSetID uint64, authorityID ed25519.PublicKeyBytes) (
+	types.OpaqueKeyOwnershipProof, error) {
+
+	combinedArg := []byte{}
+	encodedSetID, err := scale.Marshal(authSetID)
+	if err != nil {
+		return nil, fmt.Errorf("encoding set id: %w", err)
+	}
+	combinedArg = append(combinedArg, encodedSetID...)
+
+	encodedAuthorityID, err := scale.Marshal(authorityID)
+	if err != nil {
+		return nil, fmt.Errorf("encoding authority id: %w", err)
+	}
+	combinedArg = append(combinedArg, encodedAuthorityID...)
+
+	ret, err := in.Exec(runtime.GrandpaGenerateKeyOwnershipProof, combinedArg)
```

Comment:
nit use scale.NewEncoder + pre-determined buffer capacity ⚡  So we write to the same underlying memory array, without having to resize/copy

```suggestion
	const bufferSize = 8 + 32 // authSetID uint64 + ed25519.PublicKeyBytes
	buffer := bytes.NewBuffer(make([]byte, 0, bufferSize))
	encoder := scale.NewEncoder(buffer)
	err := encoder.Encode(authSetID)
	if err != nil {
		return nil, fmt.Errorf("encoding auth set id: %w", err)
	}
	err = encoder.Encode(authorityID)
	if err != nil {
		return nil, fmt.Errorf("encoding authority id: %w", err)
	}

	ret, err := in.Exec(runtime.GrandpaGenerateKeyOwnershipProof, buffer.Bytes())
```